tests/common/connections/ssh_console_conn.py

```diff
@@ -0,0 +1,109 @@
+import time
+import re
+from base_console_conn import BaseConsoleConn
+from netmiko.ssh_exception import NetMikoAuthenticationException
+
+class SSHConsoleConn(BaseConsoleConn):
+    def __init__(self, **kwargs):
+        if not kwargs.has_key("console_server_username") \
+                or not kwargs.has_key("console_server_password"):
+            raise ValueError("Either console_server__username or console_server_password is not set")
+
+        # Console via SSH connection need two groups of user/passwd
+        self.sonic_username = kwargs['username']
+        self.sonic_password = kwargs['password']
+        console_username = kwargs['console_server_username'] + r':p=' + str(kwargs['port'])
+        # Restore the default port for SSH
+        del kwargs['port']
+        kwargs['username'] = console_username
+        kwargs['password'] = kwargs['console_server_password']
```

Comment:
How do we handle console connections that require Active Directory password. This is the case for most of the new DUTs (for eg. DX10s) that are onboarded and some Fanout switches. I think we will not want to have AD creds in secrets vars file.

Reply:
Great question. The auth method used in Lab highly depends on Microsoft AD. Perhaps we could try to add a public account for test.